frontend/src/components/AccountDetailsEditContainer.vue

```diff
@@ -0,0 +1,225 @@
+              Account gespeichert
+            </div>
+          </div>
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="updateInformation()"
+          >
+            Speichern
+          </ButtonComponent>
+        </div>
+        <div :class="showadminview ? 'flex items-center pt-1' : 'collapse'">
+          <div class="flex-grow text-center">
+            <div class="text-green-500 italic" v-if="deleted">
+              Account gelöscht
+            </div>
+          </div>
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="deleteAccount()"
+          >
+            Löschen
+          </ButtonComponent>
+        </div>
+      </template>
+    </ContainerComponent>
+  </div>
+</template>
+<script setup lang="ts">
+import { ref } from 'vue';
+import TextInputComponent from './TextInputComponent.vue';
+import ButtonComponent from './ButtonComponent.vue';
+import ContainerComponent from './ContainerComponent.vue';
+import { IUser, UserClass } from '../../../model/user/IUser';
+import { AccountRestInterface } from '../restInterfaces/AccountRestInterface';
+
+const props = defineProps({
+  /**
+   * Whether the delete button is displayed
+   */
+  showadminview: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+  /**
+   * The user to display
+   */
+  user: {
+    type: Object as () => Partial<IUser>,
+    required: true,
+  },
+});
+
+const nameField = ref(props.user.name);
+const mailField = ref(props.user.mail);
+const pwField = ref('');
+const pwFieldControl = ref('');
+const disableInputShowSpinner = ref(false);
+const deleted = ref(false);
+const displayError = ref(false);
+const displaySuccess = ref(false);
+let originalUser = props.user as IUser;
+const isAdmin = ref(originalUser.type === UserClass.ADMIN);
+/**
+ * toggles the checkbox
+ */
+function toggleCheckBox(): void {
+  if (!disableInputShowSpinner.value) {
+    isAdmin.value = !isAdmin.value;
+  }
+}
+
+/**
+ * Updates Information from the User with the entered information
+ */
+async function updateInformation(): Promise<void> {
+  if (deleted.value) {
+    return;
+  }
+  displaySuccess.value = false;
+  displayError.value = false;
+  disableInputShowSpinner.value = true;
+
+  const updateUser: Partial<IUser> = { id: originalUser.id };
+
+  if (nameField.value !== '') {
+    updateUser.name = nameField.value;
+  }
+  if (mailField.value !== '') {
+    updateUser.mail = mailField.value;
+  }
+  if (pwField.value !== '') {
+    if (pwField.value === pwFieldControl.value) {
+      updateUser.password = pwField.value;
+    } else {
+      displayError.value = true;
+      disableInputShowSpinner.value = false;
+      return;
+    }
+  }
+  if (props.showadminview) {
+    updateUser.type = isAdmin.value ? UserClass.ADMIN : UserClass.REGULAR;
+  }
+  await AccountRestInterface.updateAccount(updateUser);
```

Comment:
I think we should do a check, whether the mail is correct

Reply:
regex or something would work I think, just a basic check

Reply:
i cant push regex cause of the no-useless-escape rule from lint. (`155:19  error  Unnecessary escape character: \.  no-useless-escape`)

Reply:
Do you need the escape? What regex are you trying to use?

Reply:
in regex . ist one character, but we need the . i.e in .com or max.musterman@mail.de

Reply:
Makes sense, I guess the easiest way to circumvent that, would just disabling `no-useless-escape` or at least allowing escaping dots, I'll see what I can find in the config options

Reply:
Actually, I think this behavior might even be correct, since you need to escape the `\` before the `.` in js, otherwise it'll be interpreted as a JS-escape-sequence instead of a regex one
Suggestions to fix this would include either using slashes to declare your regex (like `const regex = /REGEX_HERE/`) or escaping the `\.` to `\\.` in the RegExp constructor. At least these suggestions made it through the lint for me

Reply:
`/REGEX_HERE/` did not seem to work, replacing `\.` with `\\.` fixed the problem. just a last thing about the regex: the original regex i found allowed something like `max.mustermann@musterhausen` and did not required a tld. I changed the regex to require one. What do you think? should we allow mails without tld?

Reply:
Status update: We decided to not allow hostname. Resolving this Conversation